projects/plugins/jetpack/_inc/client/newsletter/subscriptions-settings.jsx

```diff
@@ -0,0 +1,213 @@
+				{
+					<FormFieldset>
+						<ToggleControl
+							checked={ isSubscriptionsActive && isStbEnabled }
+							disabled={
+								! isSubscriptionsActive ||
+								unavailableInOfflineMode ||
+								isSavingAnyOption( [ 'subscriptions' ] ) ||
+								! isLinked
+							}
+							toggling={ isSavingAnyOption( [ 'stb_enabled' ] ) }
+							onChange={ handleSubscribeToBlogToggleChange }
+							label={ __(
+								'Enable the “subscribe to site” option on your comment form',
+								'jetpack'
+							) }
+						/>
+						<ToggleControl
+							checked={ isSubscriptionsActive && isStcEnabled }
+							disabled={
+								! isSubscriptionsActive ||
+								unavailableInOfflineMode ||
+								isSavingAnyOption( [ 'subscriptions' ] ) ||
+								! isLinked
+							}
+							toggling={ isSavingAnyOption( [ 'stc_enabled' ] ) }
+							onChange={ handleSubscribeToCommentToggleChange }
+							label={ __(
+								'Enable the “subscribe to comments” option on your comment form',
+								'jetpack'
+							) }
+						/>
+						{ hasSubscriptionModal && (
+							<>
+								<ToggleControl
+									checked={ isSubscriptionsActive && isSmEnabled }
+									disabled={
+										! isSubscriptionsActive ||
+										unavailableInOfflineMode ||
+										isSavingAnyOption( [ 'subscriptions' ] ) ||
+										! isLinked
+									}
+									toggling={ isSavingAnyOption( [ 'sm_enabled' ] ) }
+									onChange={ handleSubscribeModalToggleChange }
+									label={ __( 'Enable subscriber modal', 'jetpack' ) }
+								/>
+								<p className="jp-form-setting-explanation">
+									{ __(
+										'Grow your subscriber list by enabling a popup modal with a subscribe form. This will show as readers scroll.',
+										'jetpack'
+									) }
+								</p>
+							</>
+						) }
+					</FormFieldset>
+				}
+			</SettingsGroup>
+			{ getSubClickableCard() }
+
+			{ ! isLinked && ! isOffline && (
+				<ConnectUserBar
+					feature="subscriptions"
+					featureLabel={ __( 'Newsletter', 'jetpack' ) }
+					text={ __( 'Connect to manage your subscriptions settings.', 'jetpack' ) }
+				/>
+			) }
+		</SettingsCard>
+	);
+}
+
+export default withModuleSettingsFormHelpers(
+	connect( ( state, ownProps ) => {
+		return {
+			isLinked: isCurrentUserLinked( state ),
+			isOffline: isOfflineMode( state ),
+			isSubscriptionsActive: ownProps.getOptionValue( 'subscriptions' ),
+			isStatsActive: ownProps.getOptionValue( 'stats' ),
+			odysseyStatsEnabled: isOdysseyStatsEnabled( state ),
+			hasSubscriptionModal: isSubscriptionModalEnabled( state ),
+			unavailableInOfflineMode: isUnavailableInOfflineMode( state, 'subscriptions' ),
+			subscriptions: getModule( state, 'subscriptions' ),
+			isWoASite: getIsWoASite( state ),
+			isStbEnabled: ownProps.getOptionValue( 'stb_enabled' ),
+			isStcEnabled: ownProps.getOptionValue( 'stc_enabled' ),
+			isSmEnabled: ownProps.getOptionValue( 'sm_enabled' ),
+		};
+	} )( SubscriptionsSettings )
+);
```

Comment:
Not a blocker, but it seems a bit unconventional to use connect() with functional components. Of course, this PR is already doing a lot to refactor these components from class to functional, which is great (and already enough work it could have been it's own PR). I would think there's some extra complexity in converting the connect() logic to hooks.

Reply:
Yeah, seems like a fine mid-step but hooks should be ultimate goal.

Rest of the settings components are still classes.